consensus/polybft/fsm_test.go

```diff
@@ -353,37 +358,52 @@ func TestFSM_BuildProposal_WithUptimeTxGood(t *testing.T) {
 	blockChainMock.On("GetSystemState", mock.Anything, mock.Anything).Return(systemStateMock).Once()
 
 	checkpointBackendMock := new(checkpointBackendMock)
-	checkpointBackendMock.On("BuildEventRoot", mock.Anything, mock.Anything).Return(types.ZeroHash, nil).Once()
+	checkpointBackendMock.On("BuildEventRoot", mock.Anything, mock.Anything).Return(eventRoot, nil).Once()
+
+	runtime := &consensusRuntime{
+		logger: hclog.NewNullLogger(),
+		config: &runtimeConfig{
+			Key:        wallet.NewKey(validators.getPrivateIdentities()[0]),
+			blockchain: blockChainMock,
+		},
+	}
 
 	fsm := &fsm{parent: parent, blockBuilder: mBlockBuilder, config: &PolyBFTConfig{}, backend: blockChainMock,
 		isEndOfEpoch:      true,
 		validators:        validators.toValidatorSet(),
 		uptimeCounter:     createTestUptimeCounter(t, nil, 10),
 		checkpointBackend: checkpointBackendMock,
 		logger:            hclog.NewNullLogger(),
-		roundInfo:         &pbft.RoundInfo{},
 	}
 
-	proposal, err := fsm.BuildProposal()
+	proposal, err := fsm.BuildProposal(currentRound)
 	assert.NoError(t, err)
 	assert.NotNil(t, proposal)
 
-	rlpBlock := stateBlock.Block.MarshalRLP()
+	block := types.Block{}
```

Comment:
Shouldn't it be a pointer?

Reply:
It does not make a difference, later on I just did & if function expects pointer